src/main/java/com/microsoft/azure/vmagent/AzureVMCloud.java

```diff
@@ -1081,17 +1081,21 @@ public FormValidation doVerifyConfiguration(
         }
 
         public ListBoxModel doFillAzureCredentialsIdItems(@AncestorInPath Item owner) {
-            return new StandardListBoxModel().withAll(
+            StandardListBoxModel listBoxModel = new StandardListBoxModel();
+            listBoxModel.add("--- Select Azure Service Principal ---", "");
+            listBoxModel.withAll(
                     CredentialsProvider.lookupCredentials(
                             AzureCredentials.class,
                             owner,
                             ACL.SYSTEM,
                             Collections.<DomainRequirement>emptyList()));
+            return listBoxModel;
         }
 
         public ListBoxModel doFillExistingResourceGroupNameItems(@QueryParameter String azureCredentialsId)
                 throws IOException, ServletException {
             ListBoxModel model = new ListBoxModel();
+            model.add("--- Select Resource Group ---", "");
```

Comment:
There is an existing `withEmptySelection` method. Do we need to call it and keep the display consolidated?